.github/workflows/cli-release.yaml

```diff
@@ -52,17 +56,22 @@ jobs:
       - build-linux-binary
       - build-os-x-binary
     runs-on: ubuntu-latest
+    permissions:
+      id-token: "write"
+      contents: "write"
+      packages: "write"
+      pull-requests: "read" 
```

Comment:
are you sure it needs all of these?
I think this workflow actually only needs `contents: write`.

Reply:
I'm not, I just yanked this from an issue on the repo for action-automatic-releases that says `contents: "write"` was insufficient. I can experiment with restricting the permissions. I couldn't find a good resource explaining what each scope actually gives you.

Reply:
ok, just read the issue (https://github.com/marvinpinto/actions/issues/340#issuecomment-983254674) and looked through the source code of the action to understand a bit better what it does.
I'm 95% confident that for us it only needs:
```
contents: write
pull-requests: read
```

The latter is because it queries PRs to get a changelog.
The id-token is definitely irrelevant here (as the issue from the other repo suggested) and packages seems also very unlikely. I guess the other person was publishing some package (npm, docker etc.) as part of the workflow and that's why they added it.

Reply:
Sweet yeah I would've thought the same. I'll test that change.